src/algorithms/machinelearning/tensorflowpredicttempocnn.cpp

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (C) 2006-2020  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+
+#include "tensorflowpredicttempocnn.h"
+
+using namespace std;
+
+namespace essentia {
+namespace streaming {
+
+const char* TensorflowPredictTempoCNN::name = essentia::standard::TensorflowPredictTempoCNN::name;
+const char* TensorflowPredictTempoCNN::category = essentia::standard::TensorflowPredictTempoCNN::category;
+const char* TensorflowPredictTempoCNN::description = essentia::standard::TensorflowPredictTempoCNN::description;
+
+
+TensorflowPredictTempoCNN::TensorflowPredictTempoCNN() : AlgorithmComposite(),
+    _frameCutter(0), _tensorflowInputTempoCNN(0), _vectorRealToTensor(0), _tensorNormalize(0),
+    _tensorTranspose(0), _tensorToPool(0), _tensorflowPredict(0), _poolToTensor(0),
+    _tensorToVectorReal(0), _configured(false) {
+
+  declareInput(_signal, 4096, "signal", "the input audio signal sampled at 11025 Hz");
+  declareOutput(_predictions, 0, "predictions", "the model predictions");
```

Comment:
What are these prediction? A more explicit description is better (e.g., `the model predictions (tempo bins activations)`)

Reply:
`TensorflowPredict`* algorithms can be used as feature extractor also depending on the node we retrieve using the `output` parameter.

What about `the output values from the model node named after 'output'`?